### ISSUE TITLE
9.1 errors

### ISSUE THREAD

**Issue by Toludin** [dandruff/xCT#195]
There seem to be several errors for xCT+ in 9.1:

```
5x xCT+\config\profile.lua:103: attempt to perform arithmetic on global 'SCHOOL_MASK_PHYSICAL' (a nil value)
[string "@xCT+\config\profile.lua"]:103: in main chunk

Locals:
ADDON_NAME = "xCT+"
addon = <table> {
 GenerateDefaultSpamSpells = <function> defined @xCT+\config\profile.lua:39
 merge_helpers = <table> {
 }
 merge2h = <table> {
 }
 merges = <table> {
 }
}
CreateComboSpellEntry = <function> defined @xCT+\config\profile.lua:26
tostring = <function> defined =[C]:-1
(*temporary) = <table> {
}
(*temporary) = <table> {
 hideConfig = true
 blizzardFCT = <table> {
 }
 showStartupText = true
 bypassCVars = false
}
(*temporary) = <table> {
 nil = <table> {
 }
}
(*temporary) = <function> defined =[C]:-1
(*temporary) = nil
(*temporary) = nil
(*temporary) = 0.750000
(*temporary) = 0.750000
(*temporary) = 0.750000
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to perform arithmetic on global 'SCHOOL_MASK_PHYSICAL' (a nil value)"
```

```
5x xCT+\modules\options-Options.lua:6215: attempt to concatenate global 'MAELSTROM' (a nil value)
[string "@xCT+\modules\options-Options.lua"]:6215: in main chunk

Locals:
build = 90100
ADDON_NAME = "xCT+"
addon = <table> {
 GenerateDefaultSpamSpells = <function> defined @xCT+\config\profile.lua:39
 DEFAULT_PET_ICON = "ability_hunter_pet_goto"
 PET_ICONS = <table> {
 }
 engine = <table> {
 }
 noop = <function> defined @xCT+\init.lua:16
 options = <table> {
 }
 merge_helpers = <table> {
 }
 merge2h = <table> {
 }
 merges = <table> {
 }
}
LSM = <table> {
 OverrideMedia = <table> {
 }
 callbacks = <table> {
 }
 LOCALE_BIT_zhTW = 8
 SetDefault = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:293
 GetDefault = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:289
 LOCALE_BIT_ruRU = 2
 UnregisterAllCallbacks = <function> defined @LibSharedMedia-3.0\CallbackHandler-1.0\CallbackHandler-1.0.lua:174
 MediaType = <table> {
 }
 LOCALE_BIT_western = 128
 SetGlobal = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:280
 GetGlobal = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:276
 List = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:266
 HashTable = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:262
 MediaTable = <table> {
 }
 RegisterCallback = <function> defined @LibSharedMedia-3.0\CallbackHandler-1.0\CallbackHandler-1.0.lua:90
 IsValid = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:258
 LOCALE_BIT_zhCN = 4
 UnregisterCallback = <function> defined @LibSharedMedia-3.0\CallbackHandler-1.0\CallbackHandler-1.0.lua:153
 Fetch = <function> defined @LibSharedMedia-3.0\LibSharedMedia-3.0\LibSharedMedia-3.0.lua:251
 DefaultMedia = <table> {
 }
 Register = <function> defined =[C]:-1
 MediaList = <table> {
 }
 LOCALE_BIT_koKR = 1
}
x = <table> {
 SetDefaultModuleLibraries = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:367
 runeIcons = <table> {
 }
 OnSpamUpdate = <function> defined @xCT+\modules\frames.lua:552
 EnableModule = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:332
 RestoreAllDefaults = <function> defined @xCT+\modules\frames.lua:1185
 IterateEmbeds = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:442
 EndConfigMode = <function> defined @xCT+\modules\frames.lua:855
 Clear = <function> defined @xCT+\modules\frames.lua:272
 Printf = <function> defined @ElvUI\Libraries\Ace3\AceConsole-3.0\AceConsole-3.0.lua:69
 CombatLogEvent = <function> defined @xCT+\modules\combattext.lua:1848
 SetDefaultModulePrototype = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:409
 player = <table> {
 }
 defaultModuleState = true
 IsEnabled = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:451
 merge = <unnamed> {
 }
 Abbreviate = <function> defined @xCT+\modules\frames.lua:299
 GetArgs = <function> defined @ElvUI\Libraries\Ace3\AceConsole-3.0\AceConsole-3.0.lua:144
 frames = <table> {
 }
 FrameTitles = <table> {
 }
 ToggleConfigMode = <function> defined @xCT+\modules\frames.lua:926
 QuickClassFrameUpdate = <function> defined @xCT+\modules\combattext.lua:677
 SetEnabledState = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:424
 UpdateBlizzardFCT = <function> defined @xCT+\modules\blizzard.lua:74
 enabledState = true
 UpdateCombatTextEvents = <function> defined @xCT+\modules\combattext.lua:137
 EnableFrameScrolling = <function> defined @xCT+\modules\frames.lua:250
 OnCombatTextEvent = <function> defined @Add

```

```
5x xCT+\config\profile.lua:40: attempt to index field 'defaults' (a nil value)
[string "@xCT+\config\profile.lua"]:40: in function `GenerateDefaultSpamSpells'
[string "@xCT+\core.lua"]:99: in function <xCT+\core.lua:89>
[string "=[C]"]: ?
[string "@ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0-13.lua"]:70: in function <...s\ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:65>
[string "@ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0-13.lua"]:498: in function `InitializeAddon'
[string "@ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0-13.lua"]:623: in function <...s\ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:615>
```

```
5x xCT+\modules\frames.lua:123: attempt to index field 'db' (a nil value)
[string "@xCT+\modules\frames.lua"]:123: in function `UpdateFrames'
[string "@xCT+\core.lua"]:155: in function <xCT+\core.lua:153>

Locals:
self = <table> {
 SetDefaultModuleLibraries = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:367
 runeIcons = <table> {
 }
 UpdateComboPointOptions = <function> defined @xCT+\core.lua:857
 OnSpamUpdate = <function> defined @xCT+\modules\frames.lua:552
 CombatStateChanged = <function> defined @xCT+\core.lua:1684
 MigratePrint = <function> defined @xCT+\core.lua:249
 EnableModule = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:332
 RestoreAllDefaults = <function> defined @xCT+\modules\frames.lua:1185
 baseName = "xCT+"
 IterateEmbeds = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:442
 EndConfigMode = <function> defined @xCT+\modules\frames.lua:855
 Clear = <function> defined @xCT+\modules\frames.lua:272
 Printf = <function> defined @ElvUI\Libraries\Ace3\AceConsole-3.0\AceConsole-3.0.lua:69
 CombatLogEvent = <function> defined @xCT+\modules\combattext.lua:1848
 SetDefaultModulePrototype = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:409
 player = <table> {
 }
 defaultModuleState = true
 IsEnabled = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:451
 GenerateColorOptions = <function> defined @xCT+\core.lua:1514
 merge = <unnamed> {
 }
 Abbreviate = <function> defined @xCT+\modules\frames.lua:299
 GetArgs = <function> defined @ElvUI\Libraries\Ace3\AceConsole-3.0\AceConsole-3.0.lua:144
 frames = <table> {
 }
 FrameTitles = <table> {
 }
 LoadAlignmentGrid = <function> defined @xCT+\modules\grid.lua:23
 ToggleConfigMode = <function> defined @xCT+\modules\frames.lua:926
 QuickClassFrameUpdate = <function> defined @xCT+\modules\combattext.lua:677
 SetEnabledState = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:424
 UpdateBlizzardFCT = <function> defined @xCT+\modules\blizzard.lua:74
 enabledState = true
 UpdateCombatTextEvents = <function> defined @xCT+\modules\combattext.lua:137
 EnableFrameScrolling = <function> defined @xCT+\modules\frames.lua:250
 OnCombatTextEvent = <function> defined @xCT+\modules\combattext.lua:511
 UpdatePlayer = <function> defined @xCT+\modules\combattext.lua:109
 IterateModules = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:437
 events = <table> {
 }
 UpdateSpamSpells = <function> defined @xCT+\core.lua:500
 UpdateItemTypes = <function> defined @xCT+\core.lua:710
 Enable = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:294
 NewModule = <function> defined @ElvUI\Libraries\Ace3\AceAddon-3.0\AceAddon-3.0.lua:235
 existingProfile = <table> {
 }
 DisableFrameScrolling = <function> defined @xCT+\modules\frames.lua:257
 TrackxCTCommand = <function> defined @xCT+\core.lua:1898
 HideConfigTool = <function> defined @xCT+\core.lua:1873
 OnAddonConfigRefreshed = <function> defined @xCT+\core.lua:1721
 defaultModuleLibraries = <table> {
 }
 ShowConfigTool = <function> defined @xCT+\core.lua:1815
 GetSpellTextureFormatted = <function> defined @xCT+\modules\combattext.lua:548
 UnregisterChatCommand = <function> defined @ElvUI\Libraries\Ace3\AceConsole-3.0\AceConsole-3.0.lua:111
 OpenxCTCommand = <function> defined @xCT+\core.lua:1730
 cvar_update = <function> defined @xCT+\modules\options.lua:149
 UpdateComboTracker = <function> defined @xCT+\core.lua:953
 OnDisable = <function> defined @xCT+\core.lua:1666
 OnEnable = <fu
```

Dumping these stacktraces here, but I'll try to see if I can find more info about what's changed in the API as well to help.

**Comment by Toludin** [comment]
SCHOOL_MASK changes:

	schoolColoring = {
		[SCHOOL_MASK_NONE]	= {a=1.0,r=1.00,g=1.00,b=1.00};
		[SCHOOL_MASK_PHYSICAL]	= {a=1.0,r=1.00,g=1.00,b=0.00};
		[SCHOOL_MASK_HOLY] 	= {a=1.0,r=1.00,g=0.90,b=0.50};
		[SCHOOL_MASK_FIRE] 	= {a=1.0,r=1.00,g=0.50,b=0.00};
		[SCHOOL_MASK_NATURE] 	= {a=1.0,r=0.30,g=1.00,b=0.30};
		[SCHOOL_MASK_FROST] 	= {a=1.0,r=0.50,g=1.00,b=1.00};
		[SCHOOL_MASK_SHADOW] 	= {a=1.0,r=0.50,g=0.50,b=1.00};
		[SCHOOL_MASK_ARCANE] 	= {a=1.0,r=1.00,g=0.50,b=1.00};
		[Enum.Damageclass.MaskNone]	= {a=1.0,r=1.00,g=1.00,b=1.00};
		[Enum.Damageclass.MaskPhysical]	= {a=1.0,r=1.00,g=1.00,b=0.00};
		[Enum.Damageclass.MaskHoly] 	= {a=1.0,r=1.00,g=0.90,b=0.50};
		[Enum.Damageclass.MaskFire] 	= {a=1.0,r=1.00,g=0.50,b=0.00};
		[Enum.Damageclass.MaskNature] 	= {a=1.0,r=0.30,g=1.00,b=0.30};
		[Enum.Damageclass.MaskFrost] 	= {a=1.0,r=0.50,g=1.00,b=1.00};
		[Enum.Damageclass.MaskShadow] 	= {a=1.0,r=0.50,g=0.50,b=1.00};
		[Enum.Damageclass.MaskArcane] 	= {a=1.0,r=1.00,g=0.50,b=1.00};

**Comment by Toludin** [comment]
Replacing line 6215 in options.lua:
`
name = "Disable |cff798BDD"..MAELSTROM,
`

with:
`name = "Disable |cff798BDD"..Enum.PowerType.Maelstrom,`

resolved the other error, but I'm not sure why only that power type had to be updated.

After making these two changes ( the SCHOOL_MASK changes and the Maelstrom change) I don't see any xCT specific errors, but I do have a bunch of null errors still. But those might be from other addons.

**Comment by Toludin** [comment]
Changing MAELSTROM to MAELSTROM_POWER in options.lua seems to have resolved that other issue, and xCT now opens for me on live. I do still have some null errors but those might be related to other addons.

**Comment by oBusk** [comment]
I've not seen this error while playing. Are you playing a Shaman? What version of the addon are you using? 

It looks to me like there is no mention of `MAELSTROM` without `_POWER` in `options.lua` since 4.5.5-Beta1 which was the version that made fixes for 9.0.5; 

https://github.com/dandruff/xCT/commit/365212b7f200f7c85a9f6ee7357bde3f09d0cb4e#diff-559a6f5cb41b13bc2a1e47e9c0c8bf737bf9f9171478eab064904ead468a8379L6283

**Comment by Toludin** [comment]
Yea I must've had an outdated/altered version then, not sure when I made those changes but if it wasn't in the main code base it must've been specific to me.

**Comment by dakaios** [comment]
with 9.1, xct is not starting anymore, config screen is empty/not loading
![grafik](https://user-images.githubusercontent.com/33833591/124349185-44543200-dbee-11eb-875e-1c2fa86b07f1.png)
 and nothing happens
already tried with all other addons deactivated - no effect
if i need to add anything please tell me where to find =)

**Comment by oBusk** [comment]
In going to guess that you are using an old version of the addon. Make sure you or your addon client installs the beta version to get the latest version that has fixes for 9.0.5 at least

**Comment by Lowenfels** [comment]
Hi, my outgoing damage is displayed the same as outgoing crit damage (with a red star) after updating to the latest version, please help.
![image](https://user-images.githubusercontent.com/86878173/124353354-a6249400-dc0e-11eb-98c3-dedf7297546a.png)

**Comment by dakaios** [comment]
> 
> 
> In going to guess that you are using an old version of the addon. Make sure you or your addon client installs the beta version to get the latest version that has fixes for 9.0.5 at least

beta version installed, same issue, config is not loading and nothing is showing

**Comment by dandruff** [comment]
Released a newer version for xCT+ on [CurseForge](https://www.curseforge.com/wow/addons/xct-plus/files/3382223) and [here](https://github.com/dandruff/xCT/releases/tag/4.5.6).

I can dedicate some more time to support this in the next few days. So please immediately report any issues you guys find :)

Have fun in Azeroth!
